detd/systemconf.py

```diff
@@ -156,13 +156,41 @@ def setup(self, interface, mapping, scheduler, stream, hints):
             return
 
         try:
-            self.vlan.setup(interface, stream, mapping)
+            self.vlan.setup_talker(interface, stream, mapping)
             self.already_configured_vids.append(stream.vid)
         except subprocess.CalledProcessError:
             self.qdisc.unset(interface)
             raise
 
+    def setup_listener(self, interface, mapping, scheduler, stream, maddress, hints):
 
+        logger.info("Setting up platform and devices")
+
+
+        try:
+            self.device.setup_listener(interface, maddress, eee="off")
+
+        except subprocess.CalledProcessError:
+            # FIXME add device restore
+            raise
+
+        #this is the important difference
+        # FIXME: consider other exceptions, e.g. TypeError
+        try:
+            # FIXME add qdisc reset
+            self.qdisc.setup(interface, mapping, scheduler, stream.base_time, hints)
```

Comment:
We do not need to configure the egress qdiscs when configuring a listener stream. This ticket provides a description of the expected configuration steps for listener streams: https://github.com/Avnu/detd/issues/4

E.g. check the sequence diagram below the phrase "This would give an idea of the call sequence".